models/experimental/functional_yolov4/reference/downsample2.py

```diff
@@ -0,0 +1,57 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+
+import torch
+import torch.nn as nn
+from models.experimental.functional_yolov4.reference.resblock import ResBlock
+
+
+class DownSample2(nn.Module):
```

Comment:
please generate the pytorch graph of DownSample2 from here and compare to the downSample2 of the reference graph from the original implementation attached here. Also include the auto-generated TTNN graph. and lets ensure all 3 graphs match as a sanity check.

Reply:
[REPO_320x320_YOLOv4_PublicRepo.gv.pdf](https://github.com/tenstorrent-metal/tt-metal/files/14938812/REPO_320x320_YOLOv4_PublicRepo.gv.pdf)

Reply:
here is a code example of generating the torch graph:

`## Example usage
model = DownSample2()

input_tensor = torch.randn(1, 3, 320, 320)  # Batch size of 1, 3 channels (RGB), 320x320 input - replace this line with the proper input size for your module
output_tensor = model(input_tensor)

model_graph = draw_graph(
    model,
    input_size=(1, 3, 320, 320),
    dtypes=[torch.float32],
    expand_nested=True,
    graph_name="DownSample2",
    depth=100,
    directory=".",
 )
 model_graph.visual_graph.render(format="pdf")`

Reply:
![downsample2_ref](https://github.com/tenstorrent-metal/tt-metal/assets/164333733/0666f263-88e9-4f90-8ba7-99d849f6a64e)
![downsample2_ttnn](https://github.com/tenstorrent-metal/tt-metal/assets/164333733/b5305ec0-fa4e-43c6-ac15-11c3f104a435)

Reply:
> ![downsample2_ref](https://private-user-images.githubusercontent.com/164333733/321586660-0666f263-88e9-4f90-8ba7-99d849f6a64e.svg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMDc1OTAsIm5iZiI6MTcxMzMwNzI5MCwicGF0aCI6Ii8xNjQzMzM3MzMvMzIxNTg2NjYwLTA2NjZmMjYzLTg4ZTktNGY5MC04YmE3LTk5ZDg0OWY2YTY0ZS5zdmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNlQyMjQxMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZmY2ZTkyNjFjNTYzYTU5ZWM0ZTk0M2IzNjMyODU2YTI5Y2NhZmVmMmU1NmYyNjQwMWVkZTVhMzc0MzBlYjM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JRmZuUr9aCdT2jjahFujzrmN2zVSzkWZvWysm8PrDBI) ![downsample2_ttnn](https://private-user-images.githubusercontent.com/164333733/321586670-b5305ec0-fa4e-43c6-ac15-11c3f104a435.svg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMDc1OTAsIm5iZiI6MTcxMzMwNzI5MCwicGF0aCI6Ii8xNjQzMzM3MzMvMzIxNTg2NjcwLWI1MzA1ZWMwLWZhNGUtNDNjNi1hYzE1LTExYzNmMTA0YTQzNS5zdmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNlQyMjQxMzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YmRmNzBjZTI1YzQ3Zjc4MGVhMWZhNWY0ZDFiNjdiNDE5MTcxZGQ4ZmY1YTE3ODkxMGI3ZDM3Yzg2M2JjMGRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.K7bC4LYkBbz9g1N4rAhOWb_YWQlP7Xk1qSdjujsr8_k)

I still dont see the pytorch graph for this one! Please use the code snipped I had generated earlier and generate the pytorch graph of your implementation, 
Please **confirm** here that you have generated the torch gerph (attache it here) AND the TTNN graph and compared them with the original graph's DS2 and DS3 portions (I had already attached the original full graph to this PR). 
The purpose of this is to ensure that we have correctly implemented the modules in torch and also to ensure the TTNN graph is a match with torch. 
We can not merge unless these are verified.

Reply:
[downsample2_pytoch.gv.pdf](https://github.com/tenstorrent/tt-metal/files/15006493/downsample2_pytoch.gv.pdf)